### DISCUSSION TITLE
HttpKernelBrowser issues reported by phpstan

### DISCUSSION THREAD

**Discussion by qdequippe** [symfony/symfony#59083] [Q&A] [answered]
👋🏼 
When upgrading to Symfony 7.2, phpstan report these issues on `src/Symfony/Component/BrowserKit/AbstractBrowser.php` in tests:

```php
$client = static::createClient();
$client->jsonRequest(...);

$response = $client->getResponse();
$content  = $response->getContent();
```

```
Call to an undefined method object::getContent(). 
```

This PR improves type information https://github.com/symfony/symfony/pull/57809

Do you think a phpstan annotation should be added like psalm directly in symfony codebase, or it must be added on our side (user side)?

**Comment by stof** [comment, marked as answer]
phpstan should be reading the `@psalm-return` annotation already.

the issue is most likely that phpstan does not properly detect `$client` as being a HttpKernelBrowser
